### PR TITLE
fix(url-store): preserve v query param for YouTube watch URLs

### PR DESCRIPTION
**Pull request by AndreiAlexandruParaschiv**
CanonicalizeUrl stripped all query parameters, collapsing every youtube.com/watch?v=<id> URL down to https://www.youtube.com/watch. Adding multiple YouTube videos in one request therefore created only the first URL; the rest hit the unique (siteId, url) constraint and came back as "Failed to create".

Preserve the `v` video id for youtube.com / www. / m. watch URLs so distinct videos map to distinct canonical URLs. Query stripping is unchanged for every other host and for non-watch YouTube paths.

### CODE REVIEW

File: src/controllers/url-store.js [hunk 1]
```diff
@@ -34,11 +34,38 @@ const MAX_URLS_PER_REQUEST = 100;
...
+ * the spacecat-audit-worker offsite-brand-presence `normalizeYoutubeUrl`, so a video added
+ * by a customer and the same video discovered by brand-presence map to the same row.
+ *
+ * @param {URL} urlObj - Parsed URL (hostname already lowercased)
+ * @returns {string|null} - `https://youtu.be/<id>`, or null when not an applicable URL
+ */
+function youTubeWatchShortForm(urlObj) {
+  const isYouTubeHost = urlObj.hostname === 'youtube.com'
+    || urlObj.hostname.endsWith('.youtube.com');
+  if (isYouTubeHost && urlObj.pathname.startsWith('/watch')) {
```
Comment (MysticatBot) [inline src/controllers/url-store.js]: issue (blocking): `pathname.startsWith('/watch')` matches any path beginning with `/watch`, not just the YouTube video player endpoint. `/watchlater` is a real YouTube path (the Watch Later playlist page). If a user submits `https://www.youtube.com/watchlater?v=somevalue`, this condition fires and produces an incorrect canonical URL.

Fix - use an exact match:
```js
if (isYouTubeHost && (urlObj.pathname === '/watch' || urlObj.pathname === '/watch/')) {
```

This is the only shape YouTube uses for video player URLs.

File: src/controllers/url-store.js [hunk 2]
```diff
@@ -34,11 +34,38 @@ const MAX_URLS_PER_REQUEST = 100;
...
+    const videoId = urlObj.searchParams.get('v');
+    if (videoId) {
```
Comment (MysticatBot) [inline src/controllers/url-store.js]: issue (blocking): `searchParams.get('v')` returns the percent-decoded value. A crafted input like `v=abc%2F..%2Fadmin` decodes to `abc/../admin`, which is then interpolated directly into `https://youtu.be/${videoId}`, producing a URL with unexpected path structure. YouTube video IDs are always `[A-Za-z0-9_-]` characters (currently 11 chars, historically variable).

Fix - validate the format before returning the short form:
```js
const videoId = urlObj.searchParams.get('v');
if (videoId && /^[A-Za-z0-9_-]+$/.test(videoId)) {
  return `https://youtu.be/${videoId}`;
}
```

Malformed `v` values fall through to the existing query-stripping path, which is the safe default.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @AndreiAlexandruParaschiv,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - two hardening gaps in the YouTube URL matching that should be tightened before merge.
**Complexity: MEDIUM** - small diff; API surface signal with user-controlled input in path construction.
_Changes: Adds YouTube watch URL special-casing to canonicalizeUrl so distinct videos produce distinct canonical URLs instead of collapsing to a single entry (2 files)._

## Must fix before merge
1. [Important] Overly broad path prefix match catches unintended YouTube paths like `/watchlater` - `src/controllers/url-store.js:54` (details inline)
2. [Important] Unsanitized video ID interpolated into output URL without format validation - `src/controllers/url-store.js:56` (details inline)

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- suggestion: Add a test for a `/watchlater?v=something` URL to lock in correct behavior once the path check is tightened - `test/controllers/url-store.test.js`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 3m 8s | <b>Cost:</b> $3.89 | <b>Commit:</b> <code>389a1621405b5d4216eabc8a35fabfce653ec75a</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @AndreiAlexandruParaschiv,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - hardening commit cleanly addresses both prior findings; no new issues.
**Complexity: MEDIUM** - small production diff; API surface signal with user-controlled input.
_Changes: Adds YouTube watch URL special-casing to canonicalizeUrl so distinct videos produce distinct canonical URLs via the youtu.be short form instead of collapsing to a single entry (2 files)._

## Previously flagged, now resolved
- Path prefix match hardened from `startsWith('/watch')` to exact `=== '/watch' || === '/watch/'` - eliminates false positives on paths like `/watchlater`
- Video ID now validated with `/^[A-Za-z0-9_-]+$/` before interpolation into the output URL - rejects encoded path-traversal payloads

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- suggestion: Add explicit test for empty `v` param (`youtube.com/watch?v=`) to document that the `+` quantifier in the regex rejects it - `test/controllers/url-store.test.js`
- suggestion: Add explicit test for absent `v` param on a `/watch` path (`youtube.com/watch?list=PLxyz`) to guard against a future refactor that drops the truthiness check (regex coerces null to the string "null" which passes) - `test/controllers/url-store.test.js`
- nit: Consider testing duplicate `v` params (`?v=abc&v=def`) to document that `searchParams.get` returns the first value - `test/controllers/url-store.test.js`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 2m 3s | <b>Cost:</b> $4.62 | <b>Commit:</b> <code>5d199a2c09d1c0e1a3c861eac75e786c9f7d63bf</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
